src/Silk.Data/MediatR/Greetings/AddGuildGreeting.cs

```diff
@@ -0,0 +1,50 @@
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Mapster;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using Remora.Results;
+using Silk.Data.DTOs.Guilds.Config;
+using Silk.Data.Entities;
+
+namespace Silk.Data.MediatR.Greetings;
+
+public static class AddGuildGreeting
+{
+    public record Request(GuildGreeting Greeting) : IRequest<Result<GuildGreeting>>;
+
+    internal class Handler : IRequestHandler<Request, Result<GuildGreeting>>
+    {
+        private readonly IDbContextFactory<GuildContext> _dbContextFactory;
+
+        public Handler(IDbContextFactory<GuildContext> dbContextFactory) 
+            => _dbContextFactory = dbContextFactory;
+
+        public async Task<Result<GuildGreeting>> Handle(Request request, CancellationToken cancellationToken)
+        {
+            await using var dbContext = await _dbContextFactory.CreateDbContextAsync(cancellationToken);
+
+            var guildConfig = await dbContext.GuildConfigs
+                                             .Include(gc => gc.Greetings)
+                                             .FirstOrDefaultAsync(gc => gc.GuildID == request.Greeting.GuildID, cancellationToken);
+            if (guildConfig is null)
+                return Result<GuildGreeting>.FromError(new NotFoundError("Guild config not found"));
+
+            var existingGreeting = guildConfig.Greetings
```

Comment:
Adding a guild greeting shouldn't possess an ID; it's generated by the database as identity.

Reply:
I know 👍, just explicitly set it to the default just in case the Id of the greeting to be created was changed/set for some reason prior to database creating the entity.